src/pages/portfolio/index.tsx

```diff
@@ -99,28 +108,57 @@ const useStyles = makeStyles((theme) => ({
   },
   title: {
     marginBottom: theme.spacing(4),
+    width: '100%',
+    zIndex: 1,
 
     [theme.breakpoints.up('sm')]: {
+      position: 'absolute',
       marginBottom: theme.spacing(3),
     },
 
     [theme.breakpoints.up('lg')]: {
+      position: 'absolute',
       margin: theme.spacing(0, 2, 4, 2),
     },
   },
+  navbarTitles: {
+    position: 'relative',
+    zIndex: 2,
+    margin: theme.spacing(-1.2, 2, 4, 2),
+  },
 }));
 
 const PortfolioPage: FC<{ data: ProjectGQL }> = ({ data }) => {
-  const projectData = data.portfolioPage.frontmatter;
   const classes = useStyles();
-  const { componentType, isDesktop } = useComponentType();
+  const defaultLabel = 'All';
+  const projectData = data.portfolioPage.frontmatter;
+  const projectLabels = projectData.projects;
+  const { componentType, isDesktop, isMobile } = useComponentType();
   const developerProfile = useDeveloperProfile();
   const [selectedProject, setSelectedProject] = useState(-1);
+  const [navbarTitle, setNavbarTitle] = useState(0);
+  const projectLabelsTitles = [defaultLabel, ...new Set(projectLabels.map((project) => project.projectLabel))];
+
+  const getNavbarTitle = (type: number) => {
+    const title: Record<number, string> = { ...projectLabelsTitles };
+    return title[type];
+  };
+
+  const projectType = getNavbarTitle(navbarTitle);
+
+  const filteredProjects = projectLabels.filter((project) => {
```

Comment:
`projectLabels` name here is confusing, because looking at the code we are filtering `projectLabels` but it returns filteredProjects, please consider renaming as suggested above 😄